generators/php/base/src/asIs/Client/RawClient.Template.php

```diff
@@ -87,60 +112,110 @@ private function buildRequest(
     }
 
     /**
+     * @param BaseApiRequest $request
+     * @param array{
+     *     headers?: array<string, string>,
+     * } $options
      * @return array<string, string>
      */
     private function encodeHeaders(
-        BaseApiRequest $request
-    ): array
-    {
+        BaseApiRequest $request,
+        array $options,
+    ): array {
         return match (get_class($request)) {
             JsonApiRequest::class => array_merge(
                 ["Content-Type" => "application/json"],
                 $this->headers,
-                $request->headers
+                $request->headers,
+                $options['headers'] ?? [],
             ),
             MultipartApiRequest::class => array_merge(
                 $this->headers,
-                $request->headers
+                $request->headers,
+                $options['headers'] ?? [],
             ),
             default => throw new InvalidArgumentException('Unsupported request type: ' . get_class($request)),
         };
     }
 
+    /**
+     * @param BaseApiRequest $request
+     * @param array{
+     *     bodyProperties?: array<string, mixed>,
+     * } $options
+     * @return ?StreamInterface
+     */
     private function encodeRequestBody(
-        BaseApiRequest $request
-    ): ?StreamInterface
-    {
+        BaseApiRequest $request,
+        array $options,
+    ): ?StreamInterface {
         return match (get_class($request)) {
-            JsonApiRequest::class => $request->body != null ? Utils::streamFor(json_encode($request->body)) : null,
+            JsonApiRequest::class => $request->body === null ? null : Utils::streamFor(
+                json_encode(
+                    $this->buildJsonBody(
+                        $request->body,
+                        $options,
+                    ),
+                )
+            ),
             MultipartApiRequest::class => $request->body != null ? new MultipartStream($request->body->toArray()) : null,
             default => throw new InvalidArgumentException('Unsupported request type: ' . get_class($request)),
         };
     }
 
+    /**
+     * @param mixed $body
+     * @param array{
+     *     bodyProperties?: array<string, mixed>,
+     * } $options
+     * @return mixed
+     */
+    private function buildJsonBody(
+        mixed $body,
+        array $options,
+    ): mixed {
+        $overrideProperties = $options['bodyProperties'] ?? [];
+        if ($body instanceof JsonSerializable) {
+            $serialized = $body->jsonSerialize();
+            return is_array($serialized)
+                ? array_merge($serialized, $overrideProperties)
+                : $serialized;
+        }
+        return is_array($body)
+            ? array_merge($body, $overrideProperties)
+            : $body;
+    }
```

Comment:
Can `bodyProperties` do deep properties?
If so, we should probably use `array_merge_recursive`

Reply:
Good call out - for now, I'd opt to keep recursive merging out to start since there might otherwise be some surprising behavior. I could imagine that this changes in the future though.